### PR TITLE
Inject special map prefix for github 'master' zip files

### PR DESCRIPTION
**Pull request by DanVanAtta**
Use a small ResourceLocationTracker location object  to contain the logic of if we need a special prefix or not when loading resources. The game previously assumed all map resources files to effectively be 'at the root'. A map downloaded from triplea-maps has a predictable format, it is named: {repoName}-master.zip, with folders "{repoName}/map".

The /map is an addition to better organize map files. We will be required to change the game code anyways to do the map rpefix, and allowing the '/map' means the current repo's do not need to be updated.

So, the Tracker finally will check the name of the zip file we are loading, if it ends with '-master.zip', then we'll assume a special path prefix, otheriwse the path prefix is empty, and things work as they did before.

---
This change is [Reviewable](https://reviewable.io/reviews/triplea-game/triplea/1164)

### CODE REVIEW

**Comment by DanVanAtta** [comment]
I suspect this is going to mess with the 'out of date' tab in the download screen. IIRC that goes off the map URL, which is going to be static soon..

File: src/games/strategy/triplea/ResourceLoader.java [hunk 1]
```diff
@@ -153,23 +152,33 @@ private ResourceLoader(final String[] paths) {
...
       }
     }
+    resourceLocationTracker = new ResourceLocationTracker(mapName, urls);
     // Note: URLClassLoader does not always respect the ordering of the search URLs
     // To solve this we will get all matching paths and then filter by what matched
     // the assets folder.
     m_loader = new URLClassLoader(urls);
   }
 
+  public void close() {
```
Comment (RoiEXLab) [inline src/games/strategy/triplea/ResourceLoader.java]: @DanVanAtta When adding a close method, this class should implement Closeable

Comment (DanVanAtta) [inline src/games/strategy/triplea/ResourceLoader.java]: Method was just moved, not added

Comment (DanVanAtta) [inline src/games/strategy/triplea/ResourceLoader.java]: Was really hard to find the class constructor, notice the constructor was moved above this method.

Comment (RoiEXLab) [inline src/games/strategy/triplea/ResourceLoader.java]: Ok didn't notice that... But still the class should implement Closable.
No other objections...

Comment (DanVanAtta) [inline src/games/strategy/triplea/ResourceLoader.java]: done, now implements Closeable

**Pull request merged**